Add cgreen-debug man page

souryogurt opened the pull request:
This pull request adds a simple man page for `cgreen-debug` script. 

I guess It might be not really useful right now because it is just an adaptation of `help2man` output format.
Unfortunately, I can't add more info in it because I'm not native English speaker :)

Rendered it looks like this:
```console
CGREEN-DEBUG(1)                  User Commands                 CGREEN-DEBUG(1)

NAME
       cgreen-debug  -  start  cgreen-runner under GDB and break at a specific
       test

SYNOPSIS
       cgreen-debug [OPTION...]  LIBRARY TEST

DESCRIPTION
       cgreen-debug is a script to  start  cgreen-runner  under  gdb,  load  a
       LIBRARY and break on a named TEST.

       -h, --help
              Print some usage information and exit.

SEE ALSO
       cgreen(5)

       The  full documentation for cgreen-debug and the Cgreen framework is in
       the Cgreen manual available at GitHub ⟨https://cgreen-devs.github.io/⟩.

cgreen 1.2.0                       May 2020                    CGREEN-DEBUG(1)
```

Closes #221

thoni56 commented:
Thanks, if I may suggest a small addition of explanation of the format the two arguments. something like this:

    LIBRARY - filename of shared library of Cgreen tests, usually `.so` or `.dll` depending on your platform.
    TEST - name of a test in that library in the format `<context>:<test_name>`

Also should there be a an addition to build/installation scripts to install this?

souryogurt commented:
Just added a description of the arguments as you said. But I added it into the `DESCRIPTION` section like in other manuals (e.g.`ln`) instead of `OPTIONS`. Is it ok? 
To be honest, I'm not really experienced in writing man pages, so your advice is very appreciated :)

```console
CGREEN-DEBUG(1)                  User Commands                 CGREEN-DEBUG(1)

NAME
       cgreen-debug  -  start  cgreen-runner under GDB and break at a specific
       test

SYNOPSIS
       cgreen-debug [OPTION...]  LIBRARY TEST

DESCRIPTION
       cgreen-debug is a script to  start  cgreen-runner  under  gdb,  load  a
       LIBRARY  and break on a named TEST.  Where LIBRARY is a filename of the
       shared library of Cgreen tests, usually .so or .dll depending  on  your
       platform.  TEST  is  the  name  of a test in that library in the format
       <context>:<test_name>.

   OPTIONS
       -h, --help
              Print some usage information and exit.

SEE ALSO
       cgreen(5) cgreen-runner(1)

       The full documentation for cgreen-debug and the Cgreen framework is  in
       the Cgreen manual available at GitHub ⟨https://cgreen-devs.github.io/⟩.
```

> Also should there be an addition to build/installation scripts to install this?

No, it is not required, because install script installs the whole directory https://github.com/cgreen-devs/cgreen/blob/6ea9e395b9b192fb454213b8ecac974ac56d6808/doc/CMakeLists.txt#L55
